### PR TITLE
add org.gabmus.unifydmin

### PR DESCRIPTION
**Pull request by GabMus**

### CODE REVIEW

File: org.gabmus.unifydmin.json [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+            "sources": [{
+                "type": "archive",
+                "url": "https://source.puri.sm/Librem5/libhandy/-/archive/v0.0.9/libhandy-v0.0.9.tar.gz",
+                "sha256": "a9f3c06eaa482efabc1d00dd2df65dc7d41253a71c0cb0402328a831a4f398a3"
+            }]
+        },
+        {
+            "name": "unifydmin",
+            "buildsystem": "meson",
+            "builddir": true,
```
Comment (bilelmoussaoui) [inline org.gabmus.unifydmin.json]: Not needed.

File: libdazzle.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "libdazzle",
```
Comment (bilelmoussaoui) [inline libdazzle.json]: Please put this module on the main one as there's no reason for splitting modules.

File: libdazzle.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  "config-opts": [
+    "--buildtype=debugoptimized"
+  ],
+  "buildsystem": "meson",
+  "sources": [
+    {
+      "type": "git",
+      "url": "https://gitlab.gnome.org/GNOME/libdazzle.git",
+      "tag": "3.31.4",
```
Comment (bilelmoussaoui) [inline libdazzle.json]: You could use a tarball here if available.

File: org.gabmus.unifydmin.json [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+    "runtime-version": "3.32",
+    "sdk": "org.gnome.Sdk",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
+        "--talk-name=org.gnome.SettingsDaemon",
+        "--talk-name=org.freedesktop.Flatpak",
+        "--filesystem=home:ro",
```
Comment (bilelmoussaoui) [inline org.gabmus.unifydmin.json]: Why does the app requires those permissions? Same for SSH.

File: org.gabmus.unifydmin.json [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+        "--filesystem=~/.ssh:ro",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "modules": [
```
Comment (bilelmoussaoui) [inline org.gabmus.unifydmin.json]: Missing cleanup to remove unused files.

File: org.gabmus.unifydmin.json (hunk 1 above)
Comment (GabMus) [reply org.gabmus.unifydmin.json]: do you mean `"builddir": true,` is not needed?

File: org.gabmus.unifydmin.json (hunk 5 above)
Comment (GabMus) [reply org.gabmus.unifydmin.json]: where should I put the cleanup and what should I cleanup?

File: org.gabmus.unifydmin.json (hunk 4 above)
Comment (GabMus) [reply org.gabmus.unifydmin.json]: - home access: it was a leftover from another project, sorry
- `~/.ssh` access, this application is an administration utility similar to cockpit in a way, that allows to control different servers all in one place. It uses ssh to connect to those servers using a library called fabric. fabric automatically logs in using ssh keys if they are available, thus the read access to `~/.ssh`. Removing it (I tried) prevents fabric from connecting, making the whole application practically useless.
- talking to org.freedesktop.Flatpak: allows to run non sandboxed commands to open a terminal with an ssh connection in your favorite terminal (user configurable). The default is xterm.

**Comment by nedrichards** [comment]
bot, build org.gabmus.unifydmin

**Comment by GabMus** [comment]
Do I need to do anything else on my end?

File: org.gabmus.unifydmin.json (hunk 1 above)
Comment (scx) [reply org.gabmus.unifydmin.json]: @GabMus 
> do you mean `"builddir": true,` is not needed?

It is default for the `meson` buildsystem.

Comment (GabMus) [reply org.gabmus.unifydmin.json]: done

**Comment by GabMus** [comment]
bot, build org.gabmus.unifydmin

File: org.gabmus.unifydmin.json (hunk 5 above)
Comment (bilelmoussaoui) [reply org.gabmus.unifydmin.json]: You should put the cleanup on either each module or on the whole manifest (prefer the first case ofc). The cleanup should remove the unneeded files that get installed (like headers, documentation or anything that will just make the size of bigger without any benefits). 
You can use `flatpak-builder` in order to build the app locally and check the files that will be part of the app itself and you can remove.

**Comment by GabMus** [comment]
bot, build org.gabmus.unifydmin

**Comment by GabMus** [comment]
I think it got stuck

**Comment by GabMus** [comment]
Building again, since the github check got stuck

**Comment by GabMus** [comment]
bot, build org.gabmus.unifydmin

**Comment by GabMus** [comment]
any news? why is this pr stuck?

**Comment by nedrichards** [comment]
@GabMus the appid doesn't follow the recommendation since it doesn't appear that anyone owns the actual domain gabmus.org https://github.com/flathub/flathub/wiki/App-Requirements#Application-ID other than that this seems *fine*, although I'm always wary about apps that need these sort of permissions.

**Comment by GabMus** [comment]
@nedrichards you're right about the domain, I just registered it right now. It should redirect to another website I own for now, but I think that works as well.

As for the permissions, the application itself is AGPL3, if you don't trust it you can audit it yourself, the repo is [here](https://gitlab.com/gabmus/unifydmin).

Besides, I already have another app submitted on flathub, it's hydrapaper. If I had any malicious intents, I'd have used hydrapaper, don't you think? Anyway, I understand the suspicion, I'd have said the same things if I were in your shoes :grin:

**Comment by nedrichards** [comment]
@GabMus that's fine on the domain front.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.gabmus.unifydmin

**Pull request closed without merging**